docs/hacking-an-agent.md

```diff
@@ -1,6 +1,200 @@
-!!!	Warning
-	Not recommended.
+## Preliminaries
 
-!!! Note
-	Coming soon.
+These instructions detail the Python code you need for running an AEA outside the `cli` tool.
 
+!!!	Note
+	You have already coded up your agent. See the <a href="../aea/skill-guide/" target=_blank>build your own skill guide</a> for a reminder.
+
+
+## Imports
+
+First, import the necessary common Python libraries and classes.
+
+``` python
+import os
+import tempfile
+import time	
+from pathlib import Path
+from threading import Thread
+import yaml
+```
+
+Then, import the application specific libraries.
+
+``` python
+from aea import AEA_DIR
+from aea.aea import AEA	
+from aea.configurations.base import ProtocolConfig
+from aea.connections.local.connection import LocalNode, OEFLocalConnection
+from aea.crypto.ledger_apis import LedgerApis
+from aea.crypto.wallet import Wallet
+from aea.mail.base import MailBox, Envelope
+from aea.protocols.base import Protocol
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.protocols.fipa.message import FIPAMessage
+from aea.protocols.fipa.serialization import FIPASerializer
+from aea.registries.base import Resources
+from aea.skills.base import Skill
+from .conftest import CUR_PATH, DummyConnection
+```
+
+
+
+## Initialise the agent
+
+Create a node.
```

Comment:
Let's use the stub connection instead:
`stub_connection = StubConnection(input_file_path='input.txt', output_file_path='output.txt')